CRM/Contact/Form/Edit/CommunicationPreferences.php

```diff
@@ -82,8 +82,33 @@ public static function buildQuickForm(&$form) {
         'greeting_type' => $greeting,
       ];
 
-      //add addressee in Contact form
+      // Add addressee in Contact form.
       $greetingTokens = CRM_Core_PseudoConstant::greeting($filter);
+
+      // Instead of showing smarty token with/o conditional logic in Drop down
+      // list, show processed token (Only in Contact Edit mode).
+      // Get Description of each greeting.
+      $greetingTokensDescription = CRM_Core_PseudoConstant::greeting($filter, 'description');
+      if ($form->_contactId) {
+        $contact = new CRM_Contact_BAO_Contact();
+        $contact->id = $form->_contactId;
+        $contact->find(TRUE);
+        $contactDetails = [];
+        CRM_Core_DAO::storeValues($contact, $contactDetails);
+        $contactDetails = [[$contact->id => $contactDetails]];
+        foreach ($greetingTokens as $key => &$emailGreetingString) {
+          if ($emailGreetingString) {
+            CRM_Contact_BAO_Contact_Utils::processGreetingTemplate($emailGreetingString,
```

Comment:
The bad news is that I think that function is one of the ones we are removing this release. If you leave it a couple of weeks we should be clearer with guidance on that - we would need a unit test though

Reply:
ok, however there is nothing mentioned in function doc about `@deprecated` for function `processGreetingTemplate` ?

Reply:
@sunilpawar no - defintely hoping to have that sorted by the time we cut 5.43 - just a couple more things to iron out

Reply:
I haven't tested, but I think this revision would switch it to the newer mechanism:

```diff
diff --git a/CRM/Contact/Form/Edit/CommunicationPreferences.php b/CRM/Contact/Form/Edit/CommunicationPreferences.php
index 7e232de0af..c9fad4bca9 100644
--- a/CRM/Contact/Form/Edit/CommunicationPreferences.php
+++ b/CRM/Contact/Form/Edit/CommunicationPreferences.php
@@ -90,17 +90,12 @@ class CRM_Contact_Form_Edit_CommunicationPreferences {
       // Get Description of each greeting.
       $greetingTokensDescription = CRM_Core_PseudoConstant::greeting($filter, 'description');
       if ($form->_contactId) {
-        $contact = new CRM_Contact_BAO_Contact();
-        $contact->id = $form->_contactId;
-        $contact->find(TRUE);
-        $contactDetails = [];
-        CRM_Core_DAO::storeValues($contact, $contactDetails);
-        $contactDetails = [[$contact->id => $contactDetails]];
+        $renderedGreetingTokens = CRM_Core_TokenSmarty::render($greetingTokens, [
+          'contactId' => $form->_contactId,
+        ]);
         foreach ($greetingTokens as $key => &$emailGreetingString) {
           if ($emailGreetingString) {
-            CRM_Contact_BAO_Contact_Utils::processGreetingTemplate($emailGreetingString,
-              $contactDetails, $form->_contactId, 'CRM_Contact_BAO_Contact'
-            );
+            $emailGreetingString = $renderedGreetingTokens[$key];
             $emailGreetingString = CRM_Core_DAO::escapeString(CRM_Utils_String::stripSpaces($emailGreetingString));
             if (!empty($greetingTokensDescription[$key])) {
               // Append description to processed greeting.
```

But that does leave another question (sort of pre-existing/tangential to this PR) -- `CRM_Contact_BAO_Contact_Utils::processGreetingTemplate` is used by a few more places  - do we switch all of these at once, switch gradually, or do some refactoring to nominally preserve the function. (AFAICS, there's nothing in `universe` which calls this function, aside from `civicrm-core`.)

Reply:
@totten your changes work perfectly. I will update the PR with new changes...